frontend/src/components/HomePage.tsx

```diff
@@ -4,26 +4,33 @@ import Button from '@material-ui/core/Button';
 import '../styles/HomePage.css';
 import axios from 'axios';
 
-class HomePage extends React.Component{
+interface State {
+  roomid: string,
+  redirect: boolean
+}
+
+class HomePage extends React.Component<{}, State> {
+  self = this;
```

Comment:
what's this for?

Reply:
I was actually wondering the same thing, but I think this was removed? My first impression was he was trying to use this on a functional component - which doesn't work (Not sure though). But then he converted the component into a class component